src/fireedge/src/server/routes/entrypoints/Api/middlawares.js

```diff
@@ -0,0 +1,187 @@
+/* ------------------------------------------------------------------------- *
+ * Copyright 2002-2023, OpenNebula Project, OpenNebula Systems               *
+ *                                                                           *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may   *
+ * not use this file except in compliance with the License. You may obtain   *
+ * a copy of the License at                                                  *
+ *                                                                           *
+ * http://www.apache.org/licenses/LICENSE-2.0                                *
+ *                                                                           *
+ * Unless required by applicable law or agreed to in writing, software       *
+ * distributed under the License is distributed on an "AS IS" BASIS,         *
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  *
+ * See the License for the specific language governing permissions and       *
+ * limitations under the License.                                            *
+ * ------------------------------------------------------------------------- */
+const { env } = require('process')
+const { DateTime } = require('luxon')
+const { httpCodes, defaults } = require('server/utils/constants')
+const { getFireedgeConfig } = require('server/utils/yml')
+const {
+  defaultWebpackMode,
+  defaultEmptyFunction,
+  defaultOpennebulaZones,
+  defaultSessionExpiration,
+} = defaults
+const { validateAuth } = require('server/utils/jwt')
+const { getDataZone } = require('server/utils/server')
+const { writeInLogger } = require('server/utils/logger')
+
+let idUserOpennebula = ''
+let userOpennebula = ''
+let passOpennebula = ''
+
+/**
+ * Validate user in global state.
+ *
+ * @param {string} user - username
+ * @param {string} token - token of user
+ * @returns {boolean} user valid data
+ */
+const userValidation = (user = '', token = '') => {
+  const nowUnix = DateTime.local().toSeconds()
+  if (
+    user &&
+    token &&
+    Array.isArray(global?.users?.[user]?.tokens) &&
+    global?.users?.[user]?.tokens?.some?.(
+      ({ token: internalToken, time }) =>
+        time > nowUnix && internalToken === token
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Avoid using `==` and `===` operators with sensitive data fields like passwords, api keys, tokens etc. as they are vulnerable to timing attack. Consider using `crypto.timingSafeEqual` or `secure-compare` for time-safe string comparison.

[Learn more](https://cwe.mitre.org/data/definitions/208.html)